### DISCUSSION TITLE
Add parameters to the request on first page visit

### DISCUSSION THREAD

**Discussion by KF42** [inertiajs/inertia#794] [Help (React)] [answered]
Hello,

I would like to persist a search between visits to a certain page, which is why I want to send the search parameters every time.

I have stored the parameters in the localstorage, but I don't know how to append them on the initial loading of the page (that is, when I open a new tab and type in the link to the page).

It works when I navigate to the tab that links to the page inside the website, since I can easily add the 'data' prop to InertiaLink components. But I don't know how to do it on the first visit.

I am unsure if this is even possible right now. Any ideas are welcome. 

Thanks for any help!

**Comment by ajnsn** [comment, marked as answer]
@KF42 Instead of using localStorage, I would recommend to store the data inside the backend session and share it as [shared data](https://inertiajs.com/shared-data#top). This way you can also set default data for the first visit or even maintain this, e.g. as per-user settings.

**Reply by reinink** [reply]
Yes, I actually do this exact thing in my own apps using a [little middleware helper](https://github.com/reinink/remember-query-strings) I created. I used to have this in Ping CRM, but removed it...just to keep that demo app as simple as possible. However, you can see how I did this [here](https://github.com/inertiajs/pingcrm/commit/0f53e5116987bcee10d68158de98a7ca2bb2bd6a) (in the commit that removed it).
